gammapy/background/models.py

```diff
@@ -168,42 +168,97 @@ def __init__(self, table, spline_kwargs=DEFAULT_SPLINE_KWARGS):
         from scipy.interpolate import UnivariateSpline
 
         self.table = table
-        self.parnames = ['amplitude', 'mean', 'stddev']
-
-        s = dict()
-        for parname in self.parnames:
-            x = self.table['x']
-            y = self.table[parname]
-            s[parname] = UnivariateSpline(x, y, **spline_kwargs)
-        self._par_model = s
-
-    def _evaluate_y(self, y, pars):
-        """Evaluate Gaussian model at a given ``y`` position."""
-        return Gaussian1D.evaluate(y, **pars)
-
-    def parvals(self, x):
-        """Interpolated parameter values at a given ``x``."""
-        x = np.asanyarray(x, dtype=float)
-        parvals = dict()
-        for parname in self.parnames:
-            par_model = self._par_model[parname]
-            shape = x.shape
-            parvals[parname] = par_model(x.flat).reshape(shape)
-
-        return parvals
-
-    def y_model(self, x):
-        """Create model at a given ``x`` position."""
-        x = np.asanyarray(x, dtype=float)
-        parvals = self.parvals(x)
-        return Gaussian1D(**parvals)
-
-    def evaluate(self, x, y):
-        """Evaluate model at a given position ``(x, y)`` position."""
-        x = np.asanyarray(x, dtype=float)
-        y = np.asanyarray(y, dtype=float)
-        parvals = self.parvals(x)
-        return self._evaluate_y(y, parvals)
+        glon = Angle(self.table['GLON']).wrap_at('180d')
+
+        splines, units = {}, {}
+
+        for column in table.colnames:
+            y = self.table[column].quantity
+            spline = UnivariateSpline(glon.degree, y.value, **spline_kwargs)
+            splines[column] = spline
+            units[column] = y.unit
+
+        self._splines = splines
+        self._units = units
+
+    def _interpolate_parameter(self, parname, glon):
+        glon = glon.wrap_at('180d')
+        y = self._splines[parname](glon.degree)
+        return y * self._units[parname]
+
+    def peak_brightness(self, glon):
```

Comment:
It's mostly a question of the API...in this case I found it difficult, because this class is very much in between other model and data class we have. I decided to  push this class more into the direction of a data class (I actually thought about renaming it to `LargeScaleComponentHGPS`...) and have a convenience layer that we also implemented for the `FluxPoints`, `EventList` or `SourceCatalog` class: the actual data is stored in a `.table` attribute and then we provide easy access to the most important quantities including unit handling etc.